File: server/api/v1/hub/hub_gateway_device_tree.go
package hub

import (
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	"github.com/flipped-aurora/gin-vue-admin/server/service"
	"github.com/gin-gonic/gin"
)

type GatewayDeviceTreeApi struct {
}

var gatewayDeviceTreeService = service.ServiceGroupApp.HubServiceGroup.GatewayDeviceTreeService

// GetGatewayDeviceTreeBody 获取网关设备列表
// @Tags Gateway
// @Summary 获取网关设备列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query GatewayDeviceVM true "获取网关设备列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /gatewaydevicetree/getgatewaydevicetreebody [get]
func (gatewayDeviceTreeApi *GatewayDeviceTreeApi) GetGatewayDeviceTreeBody(c *gin.Context) {
	gateways := gatewayDeviceTreeService.GetAllGatewayWithDevices()
	response.OkWithDetailed(gateways, "获取网关及设备成功", c)
}

File: web/src/api/hub/hub_gatewayDeviceTree.js
import service from '@/utils/request'


// @Tags Gateway Device
// @Summary 获取gatewaydevice列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "获取gatewaydevice列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /gatewaydevicetree/getgatewaydevicetreebody [get]
export const getGatewayDeviceTree = (params) => {
  return service({
    url: '/gatewaydevicetree/getgatewaydevicetreebody',
    method: 'get',
    params
  })
}
